inspirehep/modules/workflows/workflows/edit_article.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+from invenio_db import db
+
+from inspirehep.modules.workflows.tasks.actions import validate_record
+from inspirehep.modules.workflows.tasks.submission import send_robotupload
+from inspirehep.modules.workflows.utils import get_resolve_edit_article_callback_url
+from inspirehep.utils.record_getter import get_db_record
+from ..utils import with_debug_logging
+
+
+@with_debug_logging
+def change_status_to_waiting(obj, eng):
+    obj.extra_data['callback_url'] = get_resolve_edit_article_callback_url()
+    eng.wait(msg='Waiting for curation.')
+    obj.save()
+    db.session.commit()
```

Comment:
I don't think these two are ever run, the wait halts the execution, and when retaking it, it starts from the next step in the workflow, that is the validate_record one.